pkg/search/postgres/common.go

```diff
@@ -1043,6 +1043,69 @@ func RunCursorQueryForSchema[T any, PT pgutils.Unmarshaler[T]](ctx context.Conte
 	}, closer, nil
 }
 
+// RunCursorQueryForSchemaFn creates a cursor against the database
+func RunCursorQueryForSchemaFn[T any, PT pgutils.Unmarshaler[T]](ctx context.Context, schema *walker.Schema, q *v1.Query, db postgres.DB, callback func(obj PT) error) error {
+	if q == nil {
+		q = searchPkg.EmptyQuery()
+	}
+
+	query, err := standardizeQueryAndPopulatePath(ctx, q, schema, GET)
+	if err != nil {
+		return errors.Wrap(err, "error creating query")
+	}
+	if query == nil {
+		return emptyQueryErr
+	}
+
+	queryStr := query.AsSQL()
+
+	ctx, cancel := contextutil.ContextWithTimeoutIfNotExists(ctx, cursorDefaultTimeout)
+
+	tx, err := db.Begin(ctx)
+	if err != nil {
+		return errors.Wrap(err, "creating transaction")
+	}
+	defer func() {
+		defer cancel()
+		if err := tx.Commit(ctx); err != nil {
+			log.Errorf("error committing cursor transaction: %v", err)
+		}
+	}()
+
+	cursorSuffix, err := random.GenerateString(16, random.CaseInsensitiveAlpha)
+	if err != nil {
+		return errors.Wrap(err, "creating cursor name")
+	}
+	cursor := stringutils.JoinNonEmpty("_", query.From, cursorSuffix)
+	_, err = tx.Exec(ctx, fmt.Sprintf("DECLARE %s CURSOR FOR %s", cursor, queryStr), query.Data...)
+	if err != nil {
+		return errors.Wrap(err, "creating cursor")
+	}
+
+	for {
+		rows, err := tx.Query(ctx, fmt.Sprintf("FETCH %d FROM %s", cursorBatchSize, cursor))
+		if err != nil {
+			return errors.Wrap(err, "advancing in cursor")
+		}
+
+		processedRows := 0
+		for rows.Next() {
+			processedRows += 1
```

Comment:
nitpick: really a nitpick: variable name is a bit misleading. `processedRows` - should be increased after successful loop. As last action in loop. With the current implementation, it is not relevant, but maybe in the future if we do some refactoring, etc. - it can be misleading.

Reply:
Will be solved by using a command tag result